content/booklets/adacore-technologies-for-airborne-software/analysis.rst

```diff
@@ -0,0 +1,3180 @@
+
+.. index:: single: Use Case 1a: Traditional development process excluding OOT; Using Ada during the design process
+
+Using Ada during the design process
+++++++++++++++++++++++++++++++++++++
+
+.. list-table::
+   :header-rows: 1
+   :widths: 16, 84
+
+   * - |blank|
+     - **Contributions**
+
+   * - **Objectives**
+     - - Software Design Process (A-2[3,4]: 5.2.1.a)
+       - Reviews and Analyses of Source Code: Compliance with architecture (A-5[2]: 6.3.4.b), traceability (A-5[5]:6.3.4.e)
+       - Reviews and Analyses of LLR: Compatibility with target (A-5[3]: 6.3.2.c)
+       - Reviews and Analyses of architecture: Compatibility with target (A-4[10]: 6.3.3.c)"
+
+   * - **Activities**
+     - - Software Design Activities (5.2.2.a, 5.2.2.d)
+       - Software Development Process Traceability (5.5.c)
+       - Reviews and Analyses of Source Code (6.3.4)
+       - Reviews and Analyses of LLR: Compatibility with target (6.3.2)
+       - Reviews and Analyses of architecture: Compatibility with target (6.3.3)
+
+An application's design --- that is its low-level requirements and software architecture --- may be specified in many ways, combining text
+and graphics at various levels of formality. The main principle is to keep the design at a higher level of abstraction than the code:
+in particular avoiding expression of requirements as code or pseudo-code. Requirements are properties to be verified by the code and
+are not the code itself. Thus the general advice is to avoid using a programming language as the medium for expressing --- even in
+part --- the software design.
+
+.. index:: single: Ada language; Interface / implementation separation
+
+Ada, however, presents an exception to this advice. The language provides extensive facilities for capturing a program unit's
+specification (its "what") separately from the implementation (its "how"). An Ada package and an Ada subprogram each consists of a
+specification (the interface) and a body (the implementation) and a similar separation of interface from implementation is found in
+generic units, tasks, and encapsulated types.
+
+A unit's specification establishes the constraints on its usage, that is, the permitted relationships between that unit and other parts
+of the program. These are the unit's architectural properties, in contrast to its implementation. It thus makes sense for a
+significant part of the Ada specifications to be developed during the design process. An interesting effect is that the design
+elements defined as Ada specifications are easy to verify, sometimes simply by compiling the code and showing that the interface
+usages are correct.
+
+The separation of specification and implementation means that an Ada specification can have an implementation written in a different
+language, for example C. Although this may lose some of Ada's benefits, it illustrates the flexibility and relevance of the approach.
+
+Component identification
+''''''''''''''''''''''''''
```

Comment:
For this level, we use `^^^^^` instead of `'''''` in all courses. For the sake of maintenance, it would be good to stay consistent with other courses.